Source/multi.cpp

```diff
@@ -787,7 +787,7 @@ bool NetInit(bool bSinglePlayer)
 	gnTickDelay = 1000 / sgGameInitInfo.nTickRate;
 
 	for (int i = 0; i < NUMLEVELS; i++) {
-		glSeedTbl[i] = AdvanceRndSeed();
+		glSeedTbl[i] = AllocateRandomSequence(10000);

```

Comment:
Probably worth a note here that once Object and Monster seeds use non-overlapping ranges we can wind back the iterations needed when generating seeds for levels. (At least I don't think level layouts are as affected by overlapping RNG sequences...) Going from the half-range `AdvanceRndSeed()` call to using the full range of `GenerateSeed()` is the most valuable part for diversity in layouts imo.

Reply:
I think these are two separate issues. You need to allocate an RNG sequence for the item that will be generated by an object/monster seed so that two objects or monsters on the same dlvl won't use the same random numbers to produce identical or similar items. But you'd still need to allocate the RNG sequence for each dlvl so that monsters on different dlvls don't end up with overlapping sequences. Rather, burning additional random numbers to make space for item generation is only going to exacerbate the problem by increasing the number of random values used during dungeon generation. So the size of these sequences might have to be increased.

Reply:
I agree that they're separate issues, the concern here is that this is a not-insignificant amount of work being done ultimately to avoid an issue that is present during gameplay. Object, item, monster seeds are all persisted to the save file (or level delta in multi games iirc) so there's no real restriction on how they're generated, and the only reason the level seeds are being spread out like this is because the seeding of objects/items/monsters is currently using the same engine state as level generation right? We could instead generate object/item/monster seeds from a different PRNG (and discard a value from the engine used for level generation if required to keep that in sync) to completely avoid the issue. Then the extra work being done here isn't necessary and can be wound back to make starting a new game faster on low-spec devices.

Reply:
It's not just that these algorithms share engine state, but that the random value sequences share locality on the same randomized number line, so to speak. Using two different PRNGs would enable us to jump around on the number line at random, which greatly reduces the chance that two seeds will share a significant amount of locality. Discarding random values guarantees a specific amount of reduction in locality. So they're like two different solutions to the same problem of locality.

In that sense, I do agree that we can avoid doing all this work if we implement your solution, since we'll have solved the problem already.